core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -36,39 +45,52 @@ var (
 
 // upkeepStateRecord is a record that we save in a local cache.
 type upkeepStateRecord struct {
-	workID string
-	state  ocr2keepers.UpkeepState
-	block  uint64
+	WorkID          string
+	CompletionState ocr2keepers.UpkeepState
+	BlockNumber     uint64
 
-	addedAt time.Time
+	AddedAt time.Time
 }
 
 // upkeepStateStore implements UpkeepStateStore.
 // It stores the state of ineligible upkeeps in a local, in-memory cache (TODO: save in DB).
 // In addition, performed events are fetched by the scanner on demand.
 type upkeepStateStore struct {
-	lggr logger.Logger
+	// dependencies
+	orm     ORM
+	lggr    logger.Logger
+	scanner PerformedLogsScanner
 
-	cancel context.CancelFunc
+	// configuration
+	retention    time.Duration
+	cleanCadence time.Duration
 
 	mu    sync.RWMutex
 	cache map[string]*upkeepStateRecord
 
-	scanner PerformedLogsScanner
+	// service values
+	cancel context.CancelFunc
 }
 
 // NewUpkeepStateStore creates a new state store
-func NewUpkeepStateStore(lggr logger.Logger, scanner PerformedLogsScanner) *upkeepStateStore {
+func NewUpkeepStateStore(orm ORM, lggr logger.Logger, scanner PerformedLogsScanner) *upkeepStateStore {
 	return &upkeepStateStore{
-		lggr:    lggr.Named("upkeepStateStore"),
-		cache:   map[string]*upkeepStateRecord{},
-		scanner: scanner,
+		orm:          orm,
+		lggr:         lggr.Named("UpkeepStateStore"),
+		cache:        map[string]*upkeepStateRecord{},
+		scanner:      scanner,
+		retention:    CacheExpiration,
+		cleanCadence: GCInterval,
 	}
 }
 
 // Start starts the upkeep state store.
 // it does background cleanup of the cache.
 func (u *upkeepStateStore) Start(pctx context.Context) error {
+	if u.retention == 0 {
+		return fmt.Errorf("pruneDepth %d must be greater than zero", u.retention)
+	}
```

Comment:
`fmt.Errorf` is useless here since u.retention will always be zero

Reply:
True. Haha!